src/main/java/com/hazelcast/web/HazelcastInstanceLoader.java

```diff
@@ -70,26 +69,57 @@ public static HazelcastInstance createInstance(final ClusteredSessionService ses
         final String configLocation = properties.getProperty(CONFIG_LOCATION);
         final String useClientProp = properties.getProperty(USE_CLIENT);
         final String clientConfigLocation = properties.getProperty(CLIENT_CONFIG_LOCATION);
-        final boolean useClient = !isEmpty(useClientProp) && Boolean.parseBoolean(useClientProp);
-        URL configUrl = null;
-        if (useClient && !isEmpty(clientConfigLocation)) {
-            configUrl = getConfigURL(sessionService.getFilterConfig(), clientConfigLocation);
-        } else if (!isEmpty(configLocation)) {
-            configUrl = getConfigURL(sessionService.getFilterConfig(), configLocation);
-        }
-        String sessionTTLConfig = properties.getProperty(SESSION_TTL_CONFIG);
+        final boolean useClient = !isNullOrEmptyAfterTrim(useClientProp) && Boolean.parseBoolean(useClientProp);
+        final String mapName = properties.getProperty(MAP_NAME);
+        final String sessionTTL = properties.getProperty(SESSION_TTL_CONFIG);
+
+        URL configUrl = getConfigUrl(sessionService, configLocation, clientConfigLocation, useClient);
+
         if (useClient) {
-            if (sessionTTLConfig != null) {
+            if (!isNullOrEmptyAfterTrim(sessionTTL)) {
                 throw new InvalidConfigurationException("session-ttl-seconds cannot be used with client/server mode.");
             }
             boolean isSticky = Boolean.valueOf(properties.getProperty(STICKY_SESSION_CONFIG));
             return createClientInstance(sessionService, configUrl, instanceName, isSticky);
         }
-        Config config = getServerConfig(properties.getProperty(MAP_NAME), configUrl, sessionTTLConfig);
-        return createHazelcastInstance(sessionService, instanceName, config);
+
+        if (!isNullOrEmptyAfterTrim(instanceName)) {
+            HazelcastInstance existingInstance = Hazelcast.getHazelcastInstanceByName(instanceName);
+            if (existingInstance != null) {
+                LOGGER.info("Using existing HazelcastInstance [" + instanceName + "] for session replication");
+
+                if (!isNullOrEmptyAfterTrim(configLocation)) {
+                    LOGGER.warning(CLIENT_CONFIG_LOCATION + " setting ignored because existing Hazelcast instance ["
+                            + instanceName + "] is being used.");
+                }
+
+                if (!isNullOrEmptyAfterTrim(sessionTTL)) {
+                    LOGGER.warning(SESSION_TTL_CONFIG + " setting ignored because existing Hazelcast instance ["
+                            + instanceName + "] is being used.");
+                }
+
+                return existingInstance;
+            }
+        }
+
```

Comment:
`configLocation` is not relevant here, only the existing instance is returned here. We only check it a few lines before for warning the user that it is not used and that's it.